2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,242 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: PowerShellGet
+Comments Due: 7/30
+Plan to implement: Yes
+---
+
+# PowerShellGet 3.0 Module
+
+PowerShellGet has enabled an active community to publish and install PowerShell
+Modules.
+However, it has some architectural and design limitations that prevent it from
+moving forward more quickly.
+This RFC proposes some significant changes to address this.
+
+> [!NOTE]
+> There are no intentions to break existing PSGet v2 users so existing
+> users can stay on PSGet v2 and that will continue to work.
+
+## Motivation
+
+    As a PowerShell user,
+    I can discover how to install missing cmdlets,
+    so that I can be more productive.
+
+    As a PowerShellGet contributor,
+    I can more easily contribute code,
+    so that PowerShellGet evolves more quickly.
+
+## User Experience
+
+```powershell
+PS> Get-Satisfaction
+Get-Satisfaction : The term 'Get-Satisfaction' is not recognized as the name of a cmdlet, function, script file, or operable program.
+Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
+At line:1 char:1
++ Get-Satisfaction
++ ~~~
++ CategoryInfo          : ObjectNotFound: (Get-Satisfaction:String) [], CommandNotFoundException
++ FullyQualifiedErrorId : CommandNotFoundException
+
+Suggestion: You can obtain `Get-Satisfaction` by running `Install-PSResource HappyDays`.
+
+PS> Install-PSResource HappyDays
+```
+
+## Specification
+
+### Rewrite of PowerShellGet
```

Comment:
Would this still be a provider for PackageManagement?

Reply:
No, one of the big changes is complete separation from PkgMgmt, so no provider for PkgMgmt.  v2 will still work as-is.

Reply:
As I wrote in [PowerShell/PowerShellGet#475](https://github.com/PowerShell/PowerShellGet/issues/475) if you end support for `PackageManagement` or the `PowerShellGet` provider I have real concerns about the long-term viability for the plug-in (provider) model for logging/tracing ([PowerShell/PowerShell-RFC#161](https://github.com/PowerShell/PowerShell-RFC/issues/161)/[PowerShell/PowerShell-RFC#169](https://github.com/PowerShell/PowerShell-RFC/issues/169)) and credential provider.

Why should a third-party write and support a logging/tracing/credential provider when the project could be killed or strangled to death by the lack of support from the core module?

Reply:
My own opinion (ie, not an official Microsoft position) is that it's been clear for a while that PackageManagement has not been working out as originally hoped. The problem of building a unified package management experience over the many different kinds of package out there is extremely difficult, and the PackageManagement module did not solve it satisfactorily. PowerShellGet is about an order of magnitude more popular than all the other kinds of package PackageManagement tries to support put together, but implementation of it is hobbled by the PM infrastructure underneath.

Reply:
The folks who started OneGet and left it in its current state haven't been on the PowerShell team for quite some time so I don't expect any of them to author such a blog post.  I will have a blog post about PackageManagement sometime after PSConfEU.